driver-core/src/main/com/mongodb/connection/ClusterSettings.java

```diff
@@ -519,14 +527,13 @@ public String getShortDescription() {
     }
 
     private ClusterSettings(final Builder builder) {
-        // TODO: Unit test this
         if (builder.srvHost != null) {
             if (builder.srvHost.contains(":")) {
                 throw new IllegalArgumentException("The srvHost can not contain a host name that specifies a port");
             }
 
-            if (builder.hosts.get(0).getHost().split("\\.").length < 3) {
-                throw new MongoClientException(format("An SRV host name '%s' was provided that does not contain at least three parts. "
+            if (builder.srvHost.split("\\.").length < 3) {
+                throw new IllegalArgumentException(format("An SRV host name '%s' was provided that does not contain at least three parts. "
```

Comment:
Changing exception type to match all the others.

Reply:
ACK